app/controllers/AcceptMovementController.scala

```diff
@@ -53,8 +53,12 @@ class AcceptMovementController @Inject()(
       formProvider().bindFromRequest().fold(
         formWithErrors =>
           Future.successful(BadRequest(view(formWithErrors, mode))),
-        value =>
-          saveAndRedirect(AcceptMovementPage, value, mode)
+        value => {
+          val newUserAnswers: UserAnswers = request.userAnswers.filterForPages(Seq(
+            DateOfArrivalPage
+          ))
+          saveAndRedirect(AcceptMovementPage, value, newUserAnswers, mode)
+        }
```

Comment:
There's one tweak I think we should make here; I think it should only clear down the answers if the answer changes. Otherwise, if you step back through the flow without changing then it will clear everything after that point.

I.e., think about back button and then re-submit scenario.